models/sklearn/classification/evaluation.py

```diff
@@ -0,0 +1,279 @@
+
+
+def fold_evaluation(config, n_fold, X, y, class_name, tracks, process, exports_path, log_level):
+    # logger setup
+    logger = LoggerSetup(config=config,
+                         exports_path=exports_path,
+                         name="evaluation_{}".format(class_name),
+                         train_class=class_name,
+                         mode="w",
+                         level=log_level).setup_logger()
+
+    logger.info("---- EVALUATION of the model in the Folded dataset as well as in the whole dataset ----")
+    print("---- EVALUATION of the model in the Folded dataset as well as in the whole dataset ----")
+    print(colored("Evaluation and Folding..", "yellow"))
+    logger.info("number of folds set to config: {}".format(n_fold))
+    logger.debug("Sample of shuffled tracks tracks:")
+    logger.debug("{}".format(tracks[:5]))
+    logger.debug("Tracks list length: {}".format(len(tracks)))
+
+    exports_dir = "{}_{}".format(config.get("exports_directory"), class_name)
+
+    # load best model
+    load_model_params_path = os.path.join(exports_path, exports_dir, "best_model_{}.json".format(class_name))
+    with open(load_model_params_path) as model_params_file:
+        model_params_data = json.load(model_params_file)
+    
+    logger.info("Best model preprocessing step: {}".format(process))
+    models_path = FindCreateDirectory(exports_path,
+                                      os.path.join(exports_dir, "models")).inspect_directory()
+    clf = joblib.load(os.path.join(models_path, "model_grid_{}.pkl".format(process)))
+    logger.info("Best model loaded.")
+
+    # inner with K-Fold cross-validation declaration
+    random_seed = None
+    shuffle = config["k_fold_shuffle"]
+    if shuffle is True:
+        random_seed = config["seed"]
+    elif shuffle is False:
+        random_seed = None
+    print("Fitting the data to the classifier with K-Fold cross-validation..")
+    logger.info("Fitting the data to the classifier with K-Fold cross-validation..")
+    inner_cv = KFold(n_splits=n_fold,
+                     shuffle=shuffle,
+                     random_state=random_seed
+                     )
+    logger.debug("Type of X: {}".format(type(X)))
+    logger.debug("Type of y: {}".format(type(y)))
+    # tracks_fold_indexing = []
+    tracks_fold_indexing_dict = {}
+    print(tracks[0])
+    print(tracks[4])
+
+    # transformation of the data
+    features_prepared = Transform(config=config,
+                                  df_feats=X,
+                                  process=process,
+                                  train_class=class_name,
+                                  exports_path=exports_path,
+                                  log_level=log_level).post_processing()
+    logger.debug("features prepared shape: {}".format(features_prepared.shape))
+
+    accuracy_model = []
+    predictions_df_list = []
+    fold_number = 0
+    for train_index, test_index in inner_cv.split(features_prepared):
```

Comment:
this is a very long method. It would be good to split it into separate methods for loading data, processing a single split, generating images, saving files, etc

Reply:
Done. This method is splitted to sub-methods for each separate step that takes place there.